### PR TITLE
React 16.4 new Context

### PR DESCRIPTION
**Pull request by giautm**

### CODE REVIEW

File: ActionSheetProvider.js [hunk 1]
```diff
@@ -2,27 +2,31 @@ import React from 'react';
...
+    this._actionSheetRef = React.createRef();
+    this.showActionSheetWithOptions = this.showActionSheetWithOptions.bind(this);
+  }
 
-  getChildContext() {
-    return {
-      showActionSheetWithOptions: (...args) =>
-        this._actionSheetRef.showActionSheetWithOptions(...args),
-    };
+  showActionSheetWithOptions(...args) {
```
Comment (henrikra) [inline ActionSheetProvider.js]: Can you change this function to arrow function so you don't have to do this `this.showActionSheetWithOptions = this.showActionSheetWithOptions.bind(this);`

Comment (giautm) [reply ActionSheetProvider.js]: ah, use bind allow another class can be extended and reuse this method.

https://medium.com/@charpeni/arrow-functions-in-class-properties-might-not-be-as-great-as-we-think-3b3551c440b1

Comment (henrikra) [reply ActionSheetProvider.js]: We are not calling this method in loop or anything so performance isn't the issue :D

**Comment by peterpme** [comment]
Thank you for reviewing @henrikra ❤️ 

let me know when ready for merge

**Comment by giautm** [comment]
@peterpme : please check again `devDependencies` if it's not correct. :)

**Comment by henrikra** [comment]
Looks good 👍

**Comment by mgcrea** [comment]
Any chance this could be merged/release soon? I'm currently encountering this on a `react-native@0.56` projet.

**Comment by peterpme** [comment]
Hey everyone, can we use `create-react-context`
https://github.com/jamiebuilds/create-react-context

so it doesn't break for old versions?

After that I can merge and release :)

Thanks!!

cc @henrikra @giautm  @mgcrea

**Comment by peterpme** [comment]
bump @henrikra @giautm @mgcrea

**Comment by peterpme** [comment]
bump @henrikra @giautm @mgcrea

**Comment by giautm** [comment]
I close this PR because I have deleted the old fork (by a mistake). I recreated the new PR for this changes, please take a looks: https://github.com/expo/react-native-action-sheet/pull/95

**Pull request closed without merging**